lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyCompletionParticipant.java

```diff
@@ -56,32 +56,42 @@ public void onAttributeValue(String valuePrefix, ICompletionRequest request, ICo
                         .getDocumentURI());
         //getting all existing variables in current completion prefix string
         List<VariableLoc> variables = LibertyUtils.getVariablesFromTextContent(valuePrefix);
-        // value prefix will contain all existing attribute value, till the cursor
-        String variableName = valuePrefix.replace("${", "")
-                .replace("}", "");
-        StringBuilder prefixVariables = new StringBuilder();
-        String replacePrefix = "";
-        for (VariableLoc variableLoc : variables) {
-            prefixVariables.append(String.format("${%s}", variableLoc.getValue()));
-        }
+        String variablePrefix = "";
+        String completionPrefix;
+
         if (!variables.isEmpty()) {
+            // for multiple variables
             // find last variable last character and consider new variable completion from that location
             VariableLoc lastVar = variables.get(variables.size() - 1);
-            variableName = valuePrefix.substring(lastVar.getEndLoc() + 1);
-            // if last variable is started with ${
-            if (variableName.contains("${")) {
-                variableName = variableName.substring(variableName.lastIndexOf("${")).replace("${", "")
-                        .replace("}", "");
-                // add char between last variable and start of completion variable to prefix
-                replacePrefix = valuePrefix.substring(lastVar.getEndLoc() + 1, valuePrefix.lastIndexOf("${"));
+            // if last index of an existing variable is less than lastIndex of ${,
+            // this means that completion variable is started with ${
+            if (valuePrefix.lastIndexOf("${") > lastVar.getEndLoc()) {
+                // get whatever is after last ${ as variablePrefix for Completion
+                variablePrefix = valuePrefix.substring(valuePrefix.lastIndexOf("${") + 2);
+                completionPrefix = valuePrefix.substring(0, valuePrefix.lastIndexOf("${"));
+            } else {
+                variablePrefix = valuePrefix.substring(lastVar.getEndLoc() + 1);
```

Comment:
Does this handle the case where the user types `${}` and then goes inside the brackets and adds characters and triggers completion? So when they trigger completion it is `${someChars|}`. In this case, there will be one variable in the `variables` collection and the `lastIndexOf("${")` will not be > than the `lastVar.getEndLoc()`. Then this logic at line 77 will set the `variablePrefix` to the closing `}`?

Reply:
Do we have a test case for `${someVar|}`?

Reply:
Yes
![image](https://github.com/user-attachments/assets/aec56568-82fe-4ec4-b618-1fa15e64480c)